docs/reference-guide/streams/overview.md

```diff
@@ -140,3 +141,25 @@ the Kafka cluster. However, even though we cannot guarantee **exactly once**
 semantics, we tried to minimize the possibility of processing messages multiple
 times. This means committing the message offsets to the Kafka cluster happens
 right after the transaction is committed to the database.
+
+## Setting a stream offset
+
+For Kafka streams users can manually set the offset of the next consumed message 
+with a call to the query procedure `mg.kafka_set_stream_offset`
+
+```cypher
+CALL mg.kafka_set_stream_offset(stream_name, offset)
+```
+
+option|description|type|example|default
+:-:|:-:|:-:|:-:|:-:
+stream name|Name of the stream to set the offset|string|"my_stream"|/
+offset|Offset number to set|int|0|/
+
+An offset of `-1` denotes the start of the stream i.e, offset 0.
+An offset of `-2` denotes the end of the stream i.e, for each topic/partition, its logical end such that
+only the next produced message will be consumed.
+
+Note, that a stream can consume from multiple topics each of them having multiple partitions. Therefore, 
+when setting the offsets to an arbitrary number the user must be cautious that
+setting the offset of a stream internally sets all of the associated offsets of that stream(topics/partitions) to that value.
```

Comment:
:rocket: 

I was thinking about another procedure: `mg.kafka_set_topic_offset(<stream name>, <topic name>, <offset>)`. Maybe we should create a task about it or add it in the patch release. What do you think?

Reply:
That's actually a very good idea which we can definitely add it so +1 && :rocket: Moreover, this will be super useful, if/when we decouple consumers and topic partitions, that is, make consumers configurable on which partitions they consume. This will provide a concrete unit of parallelism which is required to scale consumers with topic partitions :)

Reply:
Task is created for this.